opengrok-indexer/src/main/java/org/opengrok/indexer/analysis/AnalyzerGuru.java

```diff
@@ -1110,7 +1120,7 @@ private static String readOpening(InputStream in, byte[] sig)
         in.mark(MARK_READ_LIMIT);
 
         String encoding = IOUtils.findBOMEncoding(sig);
-        if (encoding == null) {
+        if (Objects.isNull(encoding)) {
             // SRCROOT is read with UTF-8 as a default.
             encoding = StandardCharsets.UTF_8.name();
         } else {
```

Comment:
This branch for skipping the BOM can be probably avoided by using `IOUtils.createBOMStrippedReader()` below. Just an idea (needs some thought w.r.t. buffer limit).

Reply:
createBOMStrippedReader api is not skipping BOM bytes, here we have the logic to skip the BOM bytes from the stream